app/controllers/wiki_controller.rb

```diff
@@ -109,34 +108,31 @@ def new_child
 
     old_page = @page
 
-    build_wiki_page_and_content
+    build_wiki_page
 
     @page.parent = old_page
     render action: 'new'
   end
 
   # edit an existing page or a new one
   def edit
-    @page = @wiki.find_or_new_page(wiki_page_title)
-    return render_403 unless editable?
+    page = @wiki.find_or_new_page(wiki_page_title)
+    return render_403 unless editable?(page)
 
-    if @page.new_record?
-      @page.parent_id = flash[:_related_wiki_page_id] if flash[:_related_wiki_page_id]
-      @page.content = WikiContent.new(page: @page)
+    if page.new_record? && flash[:_related_wiki_page_id]
+      page.parent_id = flash[:_related_wiki_page_id]
     end
 
-    @content = @page.content_for_version(params[:version])
+    version = params[:version] if User.current.allowed_to?(:view_wiki_edits, @project)
```

Comment:
@dombesz good catch. It took me a while to figure out what was going on. `form_for` will call `to_model` if that is defined in the object passed which it was because of the `SimpleDelegator` ancestor in the `AtVersion` class. But that lead to the wrapped model being returned which always had the most recent text. I prevented this to happen now by no longer responding to `to_model`. 

I then also went ahead and simplified the form structure. With the distinction between page and content gone, that was now possible but it of course lead to other errors. But now, it should be good.